bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormContainer.java

```diff
@@ -363,4 +363,20 @@ default String getParentPagePath() {
         return null;
     }
 
+    @JsonIgnore
+    default String getFormDefinition() {
+        return null;
+    }
+
+    /**
+     * Returns whether this request is originating from edge delivery.
+     *
+     * @return whether this request is originating from edge delivery
+     * @since com.adobe.cq.forms.core.components.models.form 5.3.0
+     */
+    @Nullable
+    @JsonIgnore
+    default Boolean isEdgeDeliveryRequest() {
+        return null;
```

Comment:
Request is just a bin servlet request that can originate from anywhere. I'd prefer if we can take this logi. Inside sightly rather than introducing a public method. There is no way to distinguish requests apart of origin and referrer headers of we want to distinguish them based on their origination.

Reply:
Hi nitin, 
Inside sightly , it was not possible to fetch the request attributes. That's why i had to expose a getter api and use it from sightly.

Reply:
> Request is just a bin servlet request that can originate from anywhere. I'd prefer if we can take this logi. Inside sightly rather than introducing a public method. There is no way to distinguish requests apart of origin and referrer headers if we want to distinguish them
based on their origination.

+1, can we take this out of form container and make it part of utils. Form Container should not check such things on where the request originated from. Considering the SRP principle, I would suggest you to do it outside FormContainer

Reply:
you can make use of `use` API to extract request attribute. we can have a separate `eds.js` in container for the eds use cases.